Travis set up fails on 'npm install'

First noticed on gh-4634.

https://travis-ci.org/github/datalad/datalad/jobs/699391908#L1028

```
[...]
$ npm install grunt-contrib-qunit@>=3.0.1
/home/travis/build/datalad/datalad/node_modules/puppeteer/install.js:178
            } catch {
                    ^

SyntaxError: Unexpected token {
    at createScript (vm.js:80:10)
    at Object.runInThisContext (vm.js:139:10)
    at Module._compile (module.js:617:28)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
npm WARN enoent ENOENT: no such file or directory, open '/home/travis/build/datalad/datalad/package.json'
npm WARN ws@7.3.0 requires a peer of bufferutil@^4.0.1 but none is installed. You must install peer dependencies yourself.
npm WARN ws@7.3.0 requires a peer of utf-8-validate@^5.0.2 but none is installed. You must install peer dependencies yourself.
npm WARN datalad No description
npm WARN datalad No repository field.
npm WARN datalad No README data
npm WARN datalad No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@4.0.0 install: `node install.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the puppeteer@4.0.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-06-17T16_59_58_182Z-debug.log
travis_time:end:05bc2a5f:start=1592413194038000739,finish=1592413198203281218,duration=4165280479,event=installThe command "npm install grunt-contrib-qunit@>=3.0.1" failed and exited with 1 during .
```

Hmm, the pinning worked around an error for a while, but now that's failed:

<details>
<summary>excerpt</summary>

```
$ npm install grunt-contrib-qunit@3.1.0

> puppeteer@1.20.0 install /home/travis/build/datalad/datalad/node_modules/puppeteer
> node install.js

ERROR: Failed to download Chromium r686378! Set "PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" env variable to skip download.
{ Error: connect ECONNREFUSED 173.194.197.128:443
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1191:14)
  -- ASYNC --
    at BrowserFetcher.<anonymous> (/home/travis/build/datalad/datalad/node_modules/puppeteer/lib/helper.js:111:15)
    at Object.<anonymous> (/home/travis/build/datalad/datalad/node_modules/puppeteer/install.js:64:16)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
  errno: 'ECONNREFUSED',
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '173.194.197.128',
  port: 443 }
npm WARN enoent ENOENT: no such file or directory, open '/home/travis/build/datalad/datalad/package.json'
npm WARN datalad No description
npm WARN datalad No repository field.
npm WARN datalad No README data
npm WARN datalad No license field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! puppeteer@1.20.0 install: `node install.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the puppeteer@1.20.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2020-07-09T20_39_30_496Z-debug.log
```

</details>

https://travis-ci.org/github/datalad/datalad/jobs/706664085#L1798

So, assuming the failure sticks around, that makes figuring out what's going on here more urgent.

--- a/.travis.yml
+++ b/.travis.yml
@@ -218,8 +218,14 @@ before_install:
   - travis_retry sudo apt-get install eatmydata  # to speedup some installations
   - sudo eatmydata tools/ci/prep-travis-forssh-sudo.sh
   - tools/ci/prep-travis-forssh.sh
+  # Install nvm https://github.com/nvm-sh/nvm/blob/master/README.md
+  - curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
+  # Use install & use node v10 as grunt-contrib-qunit uses generic try catch <https://github.com/datalad/datalad/issues/4635#issue-640604347>
+  - nvm install 10 && nvm use 10
+  # Echo node and npm version for debug
+  - which node && node -v && which npm && npm -v
   # Install grunt-cli
-  - eatmydata npm install grunt-cli
+  - eatmydata npm install -g grunt-cli
   # Install optionally upstream current development so we are sure that they break nothing important for us
   - if [ ! -z "${_DL_UPSTREAM_GITANNEX:-}" ]; then sudo tools/ci/install-annex-snapshot.sh; sudo ln -s `find /usr/local/lib/git-annex.linux -maxdepth 1 -type f -perm /+x` /usr/local/bin/; else sudo eatmydata apt-get install git-annex-standalone ; fi
   # Install optionally -devel version of annex, and if goes wrong (we have most recent), exit right away
@@ -252,8 +258,7 @@ install:
   - if [ ! -z "$UNSET_S3_SECRETS" ]; then echo "usetting"; unset DATALAD_datalad_test_s3_key_id DATALAD_datalad_test_s3_secret_id; fi
   # Install grunt to test run javascript frontend tests
   - npm install grunt
-  # FIXME: Pinned because v4.0.0 caused failure (gh-4635).
-  - npm install grunt-contrib-qunit@3.1.0
+  - npm install grunt-contrib-qunit@^4.0.0
 
 script:
   # Test installation system-wide

